app/views/state_file/questions/ny_county/edit.html.erb

```diff
@@ -1,11 +1,52 @@
-<% title = t(".title", filing_year: Rails.configuration.statefile_current_tax_year  ) %>
-<% content_for :page_title, title %>
+<%
+title = t(".title", filing_year: @filing_year )
+content_for :page_title, title
+%>
+
+<% content_for :script do %>
+  <script>
+      window.addEventListener("load", () => {
+          const nycCounties = [
+              "Bronx",
+              "Kings (Brooklyn)",
+              "Manhattan (see New York)",
+              "New York (Manhattan)",
+              "Queens",
+              "Richmond (Staten Isand)"
+          ]
+          const nycResidency = <%= @nyc_residency == "full_year" %>
+
+          function displayWarnings(){
+              const value = document.querySelector("#state_file_ny_county_form_residence_county").value
+              const inNyc = nycCounties.includes(value)
+              const fullYearWarning = document.querySelector("#nyc_residency_full_year_warning")
+              const noneWarning = document.querySelector("#nyc_residency_none_warning")
+              if (nycResidency === inNyc) {
```

Comment:
i think i kind of see how this could work but would it not be more accurate to say `nycResidency && inNyc`? using `===` is confusing imo

also, very nitpicky: i think `selectedNycCounty` is a somewhat more intuitive name than `inNyc` and `nycResidency` could be clearer as `nycResidencyFullYear` (because there's a part year option)

Reply:
For booleans a and b, `a === b` != `a && b` (We don't want to show the warning when both a and b are false) - I suppose we could do `!(a ^ b)`, but I don't consider this clearer.

I've updated the variable names as suggested.